hack/lib/log/stacktrace.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+#
+# This library contains an implementation of a stack trace for Bash scripts. 
+# We assume $OS_ROOT is set
+source "${OS_ROOT}/hack/util.sh"
+
+# os::log::stacktrace::install installs the stacktrace as a handler for the ERR signal if one
+# has not already been installed and sets `set -o errtrace` in order to propagate the handler
+#
+# Globals:
+#  None
+# Arguments:
+#  None
+# Returns:
+#  - export OS_USE_STACKTRACE
```

Comment:
The Google (? it's been a while) guide asks to list external state changes that a function creates *other* than the exit code it returns with. Since this is a globally scoped variable, it seems to fit that bill. It's perhaps more useful on longer methods, though.